// svelte-debouncer/dist/index.js
"use strict";
exports.__esModule = true;
var Debouncer = /** @class */ (function () {
    function Debouncer(callback, delay, debounce_first) {
        if (delay === void 0) { delay = 1000; }
        if (debounce_first === void 0) { debounce_first = false; }
        this.callback = callback;
        this.delay = delay;
        this.debounce_first = debounce_first;
        this.timer = null;
        this.is_first_debounce = true;
    }
    Debouncer.prototype.debounce = function () {
        var _this = this;
        var args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            args[_i] = arguments[_i];
        }
        if (this.is_first_debounce && !this.debounce_first) {
            this.is_first_debounce = false;
            return;
        }
        window.clearTimeout(this.timer);
        this.timer = window.setTimeout(function () {
            _this.callback.apply(_this, args);
        }, this.delay);
    };
    return Debouncer;
}());
exports["default"] = Debouncer;
